core/contract_test.go

```diff
@@ -241,3 +246,32 @@ func TestUpdateAndGet(t *testing.T) {
 		assert.Equal(t, new(felt.Felt), sRoot)
 	})
 }
+
+func TestCommitment(t *testing.T) {
+	// Copied and pasted from contract.go:161 for testing purposes.
+	calculateContractCommitment := func(storageRoot, classHash, nonce *felt.Felt) *felt.Felt {
```

Comment:
The fact that we had to do this strengthens my thoughts on commitment calculation belonging to state. Commitment calculation was tested thru a Update call already.

Reply:
reverted